devito/arch/archinfo.py

```diff
@@ -302,6 +303,38 @@ def homogenise_gpus(gpu_infos):
     return None
 
 
+@memoized_func
+def detect_cc():
+    CUDA_SUCCESS = 0
+
+    libnames = ('libcuda.so', 'libcuda.dylib', 'cuda.dll')
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        # warning("could not load any of: " + ' '.join(libnames))
+        return None
+
+    cc_major = ctypes.c_int()
+    cc_minor = ctypes.c_int()
+    result = ctypes.c_int()
+    device = ctypes.c_int()
```

Comment:
I'm not sure I get the subtlety between these four but seems needed.

Reply:
wdym? the called underlying function expects three arguments and returns one value, all of type c_int, so in total we need four values, no?

Reply:
We need them yes.

Reply:
Better now?